### DISCUSSION TITLE
How do I update my current website to SanityPress v4.10

### DISCUSSION THREAD

**Discussion by adino007** [nuotsu/sanitypress-2023#32] [Q&A] [answered]
I have an older version of SanityPress v4.6.1 and made some heavy customizations, added new schema, modules, documents, and a lot of styling. How would one upgrade to this version without losing these changes?

**Comment by mitchuman** [comment]
@adino007
Unfortunately, I don't have the perfect solution to easily update the version, as this isn't like an NPM package where updates can be pulled rather seamlessly.

Here are some possible suggestions:
- manually copy/paste some of the key updates you may find valuable, based on the [Release Notes](https://github.com/nuotsu/sanitypress/releases) and the individual commits.
- recreate your site (looks great btw!) via a fork (if you plan to keep the repo as public) or the "use as template" action. This would give you at least better insight into the newer commits that have been pushed to the original SanityPress repo.

That said, the most recent 4.10 (and sneak peek of 4.11) are using the newest versions of Next.js and Sanity / next-sanity, but I have been experiencing some hiccups inside the Sanity studio specifically around Visual Editing and the Vercel Widget. I can't guarantee a fully working experience with the upgrade. If you do plan to do so, I would suggest going up to v4.9.0.

In the meantime, I will look into possible ways of providing a better experience to update SanityPress to newer versions. Again, I greatly appreciate your feedback and suggestion! 🖤

**Comment by adino007** [comment]
I would love to help get this project going to where you are able to update
in NPM. As i think it would benefit the overall usage of this template and
future updates being streamlined instead or rebuilt …

**Reply by mitchuman** [reply]
The tricky part is this is a template codebase with code that is meant to be modified and altered.

I am also inspired by the [shadcn/ui](https://ui.shadcn.com/) project where all code is copy/pasted.

Maybe have all modules/schema reside as an NPM package where you can import as needed.
Or perhaps a CLI might be a way to automate this process—a new field I would have to (& like to) venture into.
I'll be brainstorming on this more 🧠

**Comment by adino007** [comment]
Please update me and, if you need help I’m willing to contribute!

**Reply by nuotsu** [reply, marked as answer]
I've considered not to take the NPM package approach, as I want all parts of the _raw code_ to be editable. Making it an NPM package encapsulates the code and goes against the core concept of this project that "_everything_ should be customizable".

That said, unfortunately I can only suggest to make the updates to your project manually. You mentioned you are on v4.6. There have been breaking changes since >4.12 and >5.0 so I suggest making an upgrade to 4.11.0. Here's a [helpful link (4.6.0...4.11.0)](https://github.com/nuotsu/sanitypress/compare/4.6.0...4.11.0) to compare changes.

>4.12 includes the Next.js 15 updates, which has major changes to how data is fetched.
>5.0 includes the update to Sanity's new Live API (new fetch functions).

If it's mainly styles and new modules that you have customized, I will suggest starting with a new 5.x version and migrating those changes onto the new template.

**Comment by adino007** [comment]
I mostly created Sanity Schema Components and Next.js Modules and Dynamic Data to populate grids and dynamic pages for Events, Artists, and Venues…

Check it out: koa-nextjs-sanity-website-v2.vercel.app <https://koa-nextjs-sanity-website-v2.vercel.app/>

I would love to upgrade to version 5 from ***@***.***… Last time I wasn’t too far into development so I just redid the website using a newer version, but I would rather do the code change like you said as I didn’t know that was a feature in GitHub!

Would love to hear a recommendation to upgrading to version 5!

**Reply by nuotsu** [reply]
Again, there's too many breaking changes bumping from a 4.x version to 5.x.

I would start a new 5.1.0 site (latest version) and copy over mainly the CSS and any modules from the previous site onto the new one. I suggest doing one change at a time, test, and commit your changes (so easier to revert and/or track your new edits).

Awesome site btw! 🔥

**Comment by adino007** [comment]
Thank you for your help, and the cool website is thanks to your work! I will transfer using a new project… Hopefully it will align easily!
